apps/emqx_bridge/test/emqx_bridge_mqtt_SUITE.erl

```diff
@@ -954,12 +957,11 @@ assert_mqtt_msg_received(Topic, Payload) ->
     receive
         {deliver, Topic, #message{payload = Payload}} ->
             ct:pal("Got mqtt message: ~p on topic ~p", [Payload, Topic]),
-            ok;
-        Msg ->
-            ct:pal("Unexpected Msg: ~p", [Msg]),
-            assert_mqtt_msg_received(Topic, Payload)
-    after 100 ->
-        ct:fail("timeout waiting for ~p on topic ~p", [Payload, Topic])
+            ok
+    after 300 ->
```

Comment:
Doesn't this introduce another potentially flaky test?

A better way to test this could be to send another message, a stopper on the same topic, collect messages from the topic until stopper message is received and assert on the collected one(s).

Or some snabbkaffe solution?